paddle/fluid/operators/split_op.cc

```diff
@@ -78,9 +78,18 @@ class SplitOp : public framework::OperatorWithKernel {
 
 #ifdef PADDLE_WITH_MKLDNN
     if (this->CanMKLDNNBeUsed(ctx, input_data_type)) {
-      return framework::OpKernelType(input_data_type, ctx.GetPlace(),
-                                     framework::DataLayout::kMKLDNN,
-                                     framework::LibraryType::kMKLDNN);
+      // OneDNN uses blocking format, which cannot be always
+      // supported with reorders, because if blocked dimension is not divisible
+      // by
+      // 8 or 16(depending on which blocking format is used) submemory cannot be
+      // created, so in that scenario a fallback is needed
+      auto tmp_md = dnnl::memory::desc(
+          framework::vectorize(ctx.Input<Tensor>("X")->dims()),
+          dnnl::memory::data_type::f32, ctx.Input<Tensor>("X")->format());
```

Comment:
Will it always be `dnnl::memory::data_type::f32` when checking inner_nblks ?

Reply:
This datatype doesn't matter. I only need inner_nblks and it will be the same for any datatype. I have asked oneDNN team if I can check if a format_tag is blocking in any simpler way, but they said that there is no easy way to do that, so I am constructing temporary memory descriptor to get inner_nblks